packages/fxa-settings/src/components/Auth/index.tsx

```diff
@@ -0,0 +1,51 @@
+import React, {useContext} from 'react';
+import AuthClient from 'fxa-auth-client/browser';
+import { useQuery, gql } from '@apollo/client';
+import { useAwait } from 'fxa-react/lib/hooks';
+import { AccountData } from '../AccountDataHOC/gql';
+import { sessionToken } from '../../lib/cache'
+
+export const GET_PASSWORD_INFO = gql`
+  query GetPasswordInfo {
+    account {
+      emails {
+        email
+        isPrimary
+      }
+    }
+    sessionToken @client
+    oldPassword @client
+    newPassword @client
+  }
+`
+
+export interface AuthContextValue {
+  client?: AuthClient
+};
+
+export const AuthContext = React.createContext<AuthContextValue>({})
+export {AuthClient}
+
+export function useAuth() {
+  const {client} = useContext(AuthContext)
+  if (!client) {
+    throw new Error('Are you forgetting an AuthContext.Provider?')
+  }
+  return client!;
+}
+
+export function usePasswordChanger() {
```

Comment:
This isn't a component so it will move somewhere else. I'm still trying to decide if hooks are the right style for these operations.